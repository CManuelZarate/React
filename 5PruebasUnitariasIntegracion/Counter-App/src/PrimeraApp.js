import React from 'react';//imp por defecto e independiente
import PropTypes from 'prop-types';

//FC

const PrimeraApp =( { saludo, subtitulo } ) => {
    
    if (!saludo){
        throw new Error ('El saludo es necesario');
    }
    return (
    <>
        <h1> { saludo } </h1>
        <pre>  {/*  JSON.stringify(saludo,null,3) */} </pre>
        <p>{ subtitulo }</p>
    </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'soy un subtiutlo'
}

export default PrimeraApp;