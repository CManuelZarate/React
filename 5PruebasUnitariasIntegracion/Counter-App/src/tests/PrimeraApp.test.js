import React from 'react';
import '@testing-library/jest-dom';//para q aparezca la ayuda
import { shallow } from 'enzyme';


//import {render} from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';

describe('Pruebas en <PrimeraApp />', () => { 


    /* test('debe mostrar el mensaje "Hola mundo"', () => { 

        const saludo = 'Hola mundo';//msg evaluar

        const {getByText} = render( <PrimeraApp saludo={saludo}/> ) //producto renderizado de lo quiero mostrar,,dessetruturamos el wraper y obtene el get..

        expect( getByText(saludo) ).toBeInTheDocument(); //pregunto si existe el saludo en ese comp

        
     }) */

     test('debe de mostrar el <PrimeApp /> correctamente', () => { 
        const saludo = 'Hola mundo'
        const wrapper = shallow( <PrimeraApp saludo={saludo} /> )

        expect( wrapper ).toMatchSnapshot();
      });

      test('debe de mostrar el subtitulo neviado por porps', () => { 

        const saludo = 'Hola mundo';
        const subtitulo = 'soy un subtitulo';
        const wrapper = shallow( 
        <PrimeraApp 
            saludo={saludo} 
            subtitulo={ subtitulo }

        /> );

        const textoParrafo = wrapper.find('p').text();//busco el parrafo, #id
        console.log(textoParrafo);

        expect( textoParrafo ).toBe( subtitulo );
       })

 })