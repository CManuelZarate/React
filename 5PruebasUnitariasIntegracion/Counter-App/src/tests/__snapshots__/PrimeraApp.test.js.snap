// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe de mostrar el <PrimeApp /> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola mundo
     
  </h1>
  <pre>
      
     
  </pre>
  <p>
    soy un subtiutlo
  </p>
</Fragment>
`;
